common/functions/src/aggregates/aggregate_function_state.rs

```diff
@@ -100,25 +100,21 @@ impl From<StateAddr> for usize {
 /// # Safety
 /// layout must ensure to be aligned
 pub unsafe fn get_layout_offsets(funcs: &[AggregateFunctionRef]) -> (Layout, Vec<usize>) {
-    let mut align_aggregate_states = 0;
-    let mut total_size_aggregate_states = 0;
-
-    let mut offsets_aggregate_states = Vec::with_capacity(funcs.len());
-    for i in 0..funcs.len() {
-        offsets_aggregate_states.push(total_size_aggregate_states);
-        let layout = funcs[i].state_layout();
-
-        total_size_aggregate_states += layout.size();
-        align_aggregate_states = align_aggregate_states.max(layout.align());
-        // Not the last aggregate_state
-        if i + 1 < funcs.len() {
-            let next_layout = funcs[i + 1].state_layout();
-            total_size_aggregate_states = (total_size_aggregate_states + next_layout.align() - 1)
-                / next_layout.align()
-                * next_layout.align();
-        }
+    let mut max_align = 0;
+    let mut total_size = 0;
+
+    let mut offsets = Vec::with_capacity(funcs.len());
+    for func in funcs {
+        let layout = func.state_layout();
+        let align = layout.align();
+
+        total_size = (total_size + align - 1) / align * align;
+
+        offsets.push(total_size);
+
+        max_align = max_align.max(align);
+        total_size += layout.size();
```

Comment:
「size: 2, align: 8」

total_size will be `3`, max_align will be `8` ?

Reply:
Hmmm? the result should be `total_size=2, max_align=8`?

That's just what the former version does in a simpler way.

Reply:
Did you mean that the size should not be smaller than align or something else?

Reply:
> Hmmm? the result should be total_size=2, max_align=8? That's just what the former version does in a simpler way.

Aha, you are right, I was wrong.